src/workclaimer.js

```diff
@@ -25,22 +26,22 @@ class HintPoller {
 
   requestClaim(count, aborted) {
     assert(!this.destroyed, 'requestClaim() called after destroy()');
-    // Make a request for count tasks
-    let request = null;
-    let result = new Promise((resolve, reject) => {
-      request = {resolve, reject, count};
+    return new Promise((resolve, reject) => {
+      // Make a request for count tasks
+      let request = {resolve, reject, count};
+      this.requests.push(request);
+
+      // Remove request if aborted
+      aborted.then(() => {
+        // Remove request from requests, but modifying the requests array
+        _.pull(this.requests, request);
+        // Resolve request empty array
+        request.resolve([]);
+      }).catch(err => process.nextTick(() => this.parent.emit('error', err)));
```

Comment:
@imbstack, do you agree that this would likely show up in sentry, I'm like 30%  sure I've done that right here and below...